mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobDefinition.java

```diff
@@ -50,7 +60,7 @@ public class JobDefinition {
     private final SchedulingInfo schedulingInfo;
     private final DeploymentStrategy deploymentStrategy;
     private final int withNumberOfStages;
-    private List<Label> labels;
+    private Map<String, Label> labels; // Map label->name to label instance.
```

Comment:
This representation feels weird, given that a label is essentially a key-value pair. However, if this is done for dedup purposes, then we might as well convert this into a Map<String, String> and (key, value) tuple representing the label.

Reply:
It's to ensure the unqiueness of per-label-name to label instance. I feel that converting 'Label' instance is not necessary in this case (String/Key, Object instance) seems to be a fairly normal way to track?